llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -3379,6 +3379,149 @@ Value *InstCombinerImpl::foldAndOrOfICmps(ICmpInst *LHS, ICmpInst *RHS,
   return foldAndOrOfICmpsUsingRanges(LHS, RHS, IsAnd);
 }
 
+struct BitFieldAddBitMask {
```

Comment:
Please add some comments for these structs.

Reply:
sorry about that :sweat: